nvme.c

```diff
@@ -55,7 +55,15 @@
 #include <sys/mman.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/random.h>
+
+#ifdef HAVE_SYS_RANDOM
+	#include <sys/random.h>
+#else
+	#include <sys/syscall.h>
+	#include <linux/random.h>
+	#define getrandom(buf, buflen, flags) \
+		syscall(SYS_getrandom, buf, buflen, flags)
```

Comment:
Hmm. Caling a syscall directly is always a bit awkward. Wouldn't it be better to read from /dev/urandom (or /dev/random if /dev/urandom doesn't exist)?
Then we would avoid having to use the syscall directly (which always has a chance of failing as there is no guarantee that 'SYS_getrandom' is defined) and would at least _attempt_ to be portable.

Reply:
I added reading /dev/urandom & random